dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/authorization/providers/builtin/DataverseUserPage.java

```diff
@@ -515,73 +516,78 @@ private String getRoleStringFromUser(AuthenticatedUser au, DvObject dvObj) {
-                    break;
-
-                case CHECKSUMIMPORT:
-                    userNotification.setTheObject(datasetVersionService.getById(userNotification.getObjectId()));
-                    break;
-
-                default:
-                    if (!isBaseNotification(userNotification)) {
-                        Optional.ofNullable(datasetDao.find(userNotification.getObjectId()))
+                        break;
+                    case CREATEDV:
+                        userNotification.setTheObject(dataverseDao.find(userNotification.getObjectId()));
+                        break;
+
+                    case REQUESTFILEACCESS:
+                        DataFile file = fileService.find(userNotification.getObjectId());
+                        userNotification.setTheObject(file.getOwner());
+                        break;
+                    case GRANTFILEACCESS:
+                    case REJECTFILEACCESS:
+                        userNotification.setTheObject(datasetDao.find(userNotification.getObjectId()));
+                        break;
+
+                    case MAPLAYERUPDATED:
+                    case CREATEDS:
+                    case SUBMITTEDDS:
+                    case PUBLISHEDDS:
+                    case RETURNEDDS:
+                        userNotification.setTheObject(datasetVersionService.getById(userNotification.getObjectId()));
+                        break;
+
+                    case MAPLAYERDELETEFAILED:
+                        userNotification.setTheObject(fileService.findFileMetadata(userNotification.getObjectId()));
+                        break;
+
+                    case CREATEACC:
+                        userNotification.setTheObject(userNotification.getUser());
+                        break;
+
+                    case CHECKSUMFAIL:
+                        userNotification.setTheObject(datasetDao.find(userNotification.getObjectId()));
+                        break;
+
+                    case FILESYSTEMIMPORT:
+                        userNotification.setTheObject(datasetVersionService.getById(userNotification.getObjectId()));
+                        break;
+
+                    case CHECKSUMIMPORT:
+                        userNotification.setTheObject(datasetVersionService.getById(userNotification.getObjectId()));
+                        break;
+
+                    default:
+                        if (!isBaseNotification(userNotification)) {
+                            Optional.ofNullable(datasetDao.find(userNotification.getObjectId()))
                                 .ifPresent(userNotification::setTheObject);
-                    }
-                    break;
+                        }
+                        break;
+                }
+                //TransactionRolledBackException thrown when object not found with getById
+            } catch (Exception e) {
+                logger.log(Level.WARNING, e.getMessage());
```

Comment:
I don't think we should handle it like that.

First of all: we would log many times something that is normal situation for us.
That is - object connected to notification no longer exists. 

Second: if something will go wrong we will interpret that situation as if object connected to notification not exists. 

Third: message only would probably not give us explanation what really happen. For example what we can deduce from message: `Exception thrown from bean`?

Reply:
Ok, maybe better solution would be not to throw any exception at service level because it breaks transaction manager